cgi-bin/dashboard-aii.cgi

```diff
@@ -0,0 +1,335 @@
+
+  # find all profiles in directory
+  push @profiles,map { s/\.json$//; s/^$profile_prefix//; $_ .=''; } sort(grep(/\.json$/, readdir(DIR)));
+
+  closedir(DIR);
+  # Load the configurations list
+  opendir(DIR, $pxelinux_dir);
+  @cfg = sort(grep(/(\.cfg$)|(default)/, readdir(DIR)));
+  closedir(DIR);
+}
+
+#########################################################################
+sub GetHosts {
+#########################################################################
+
+  my (@all, $k, $json);
+
+  for $k (@profiles) {
+
+    my $hostname = $k;
+
+    my ($hexaddr,$dotaddr) = GetHexAddr($hostname);
+
+    my $link = readlink("$pxelinux_dir/$hexaddr");
+    my $existing_cfg = $link ? $link : "";
+
+    push @all ,{'hostname' => $hostname, 'hexaddr' => $hexaddr, 'dotaddr' => $dotaddr, 'bootcfg' => $existing_cfg };
+
+  }
+
+  my $json =  JSON::XS->new->pretty->encode({hosts => \@all, available_cfg => \@cfg});
+
+  print "Content-type: application/json\n\n$json";
+
+}
+
+#########################################################################
+sub GetProfile {
+#########################################################################
+
+  if ($_[0] =~ /^(.*)$/) {
+    $_[0] = $1;
+    my $hostname = $_[0];
+    my $file = "$profiles_dir/$profile_prefix$hostname.json";
+
+      my $json_text = do {
+        open(my $json_fh, "<:encoding(UTF-8)", $file) or die("Can't open $file: $!\n");
+        local $/;
+        <$json_fh>
+      };
+
+      my $profile = JSON::XS->new->decode($json_text);
+      
+      #Remove sensitive sections
+      $profile->{'software'} = ();
+
+      my $json =  JSON::XS->new->pretty->allow_nonref->encode($profile);
+
+      print "Content-type: application/json\n\n$json";
+    }
+}
+
+#########################################################################
+sub Configure {
+#########################################################################
+
+  $ENV{PATH}="/bin:/usr/bin:/sbin:/usr/bin:/usr/sbin";
+
+  my $result = '';
+  print "Content-type: text/plain\n\n";
+
+  $_[0] =~ /^(configure|install|reinstall|boot)$/;
+  $_[0] = $1;
+
+  $_[1] =~ /^(.*)$/;
+  $_[1] = $1;
+
+  my $action = $_[0];
+  my $hostname = $_[1];
+
+  if ($action eq 'reinstall') {
+    my $ok = system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--remove", $hostname);
+    $ok = system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--configure", $hostname);
+    $ok = system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--install", $hostname);
+    print "Failed to reinstall $node using aii-shellfe<br>" if ($ok != 0);
+  }
+  else {
+    if (system ("/usr/bin/sudo", "/usr/sbin/aii-shellfe", "--$action", $hostname) != 0) {
+        print "Failed to $action $node using aii-shellfe<br>";
+      }
+  }
+
+}
+
+#########################################################################
+sub GetStats {
+#########################################################################
+
+  my (%all, %result, $k, $i, $json, $hostname, $value);
+
```

Comment:
@samary i'm going to stop here. some of the above comments also apply below, i'll continue when you made your first round of cleanup :smile: 
but i saw the demo, very nice work!